### ISSUE TITLE
Fix retirement test

### ISSUE THREAD

**Issue by tilk** [kuznia-rdzeni/coreblocks#556]
It was missed in review that test_rand in retirement tests has a yield outside of test processes.

**Comment by piotro888** [comment]
I found it and unfortunately it is due to `# TODO: mocking really seems to dislike nonexclusive methods for some reason`. In new code `rob_peek` is called from two transactions and simulation hangs up.

I also fixed other bugs, but the test wont run without this one solved.

**Comment by piotro888** [comment]
Even manual mocking like this:
```python
 def peek_process():
            yield from retc.mock_rob_peek.enable()
            yield Passive()
            while self.submit_q:
                yield from retc.mock_rob_peek.call_init(self.submit_q[0])
                yield
```
fails when nonexclusive method is called multiple times in dut

**Comment by piotro888** [comment]
> `# TODO: mocking really seems to dislike nonexclusive methods for some reason`.

Fixed in #558
